### ISSUE TITLE
Incorrect accessibility semantics on the `Table` widget

### ISSUE THREAD

**Issue by chrisnorman7** [flutter/flutter#100056]
Hi,
When using the `Table` widget, accessibility information doesn't seem to be generated.

In particular when running the below code sample on Flutter Web, I cannot navigate the resulting table with standard screen reader commands (using NVDA), and NVDA does not know there is a table on the page.

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('People'),
      ),
      body: Table(
        children: const [
          TableRow(
            children: [
              TableCell(child: Text('First Name')),
              TableCell(child: Text('Surname')),
              TableCell(child: Text('Age'))
            ],
          ),
          TableRow(
            children: [
              Focus(child: TableCell(child: Text('Chris'))),
              Focus(child: TableCell(child: Text('Norman'))),
              Focus(child: TableCell(child: Text('32')))
            ],
          ),
          TableRow(children: [
            Focus(child: TableCell(child: Text('Mother'))),
            Focus(child: TableCell(child: Text('Earth'))),
            Focus(child: TableCell(child: Text('Pretty old')))
          ])
        ],
      ),
    );
  }
}
```

```shell
[ÔêÜ] Flutter (Channel master, 2.11.0-0.0.pre.871, on Microsoft Windows [Version 10.0.22000.556], locale en-GB)
    ÔÇó Flutter version 2.11.0-0.0.pre.871 at C:\Users\chris\flutter
    ÔÇó Upstream repository https://github.com/flutter/flutter.git
    ÔÇó Framework revision 4f4c7816c3 (6 hours ago), 2022-03-14 07:35:07 +0100
    ÔÇó Engine revision 6239bfb884
    ÔÇó Dart version 2.17.0 (build 2.17.0-182.0.dev)
    ÔÇó DevTools version 2.11.1

[ÔêÜ] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc2)
    ÔÇó Android SDK at C:\Users\chris\AppData\Local\Android\sdk
    ÔÇó Platform android-31, build-tools 31.0.0-rc2
    ÔÇó Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    ÔÇó Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    ÔÇó All Android licenses accepted.

[ÔêÜ] Chrome - develop for the web
    ÔÇó Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[ÔêÜ] Visual Studio - develop for Windows (Visual Studio Enterprise 2022 17.1.0)
    ÔÇó Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Enterprise
    ÔÇó Visual Studio Enterprise 2022 version 17.1.32210.238
    ÔÇó Windows 10 SDK version 10.0.22000.0

[ÔêÜ] Android Studio (version 2020.3)
    ÔÇó Android Studio at C:\Program Files\Android\Android Studio
    ÔÇó Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    ÔÇó Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    ÔÇó Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[ÔêÜ] VS Code, 64-bit edition (version 1.65.1)
    ÔÇó VS Code at C:\Program Files\Microsoft VS Code
    ÔÇó Flutter extension version 3.27.0

[ÔêÜ] Connected device (4 available)
    ÔÇó Windows (desktop)       ÔÇó windows ÔÇó windows-x64     ÔÇó Microsoft Windows [Version 10.0.22000.556]
    ÔÇó Windows (UWP) (desktop) ÔÇó winuwp  ÔÇó windows-uwp-x64 ÔÇó 
    ÔÇó Chrome (web)            ÔÇó chrome  ÔÇó web-javascript  ÔÇó Google Chrome 99.0.4844.51
    ÔÇó Edge (web)              ÔÇó edge    ÔÇó web-javascript  ÔÇó Microsoft Edge 98.0.1108.56

[ÔêÜ] HTTP Host Availability
    ÔÇó All required HTTP hosts are available

ÔÇó No issues found!
```

Thanks in advance.

**Comment by maheshmnj** [comment]
cc: @cbracken

**Comment by chrisnorman7** [comment]
@maheshmnj 
Thank you for confirming. Semi unrelated: What software are you using to make those screen recordings please? They might help me explain my issues better.

Thanks again.

**Comment by maheshmnj** [comment]
@chrisnorman7 I am using the built-in game bar feature on windows (activated using Win + g) with a microphone enabled when recording.

**Comment by chrisnorman7** [comment]
Nice, thank you.

Take care,

Chris Norman

**Comment by UNIcodeX** [comment]
I work for a school and am working on adding Semantics to support our blind users. When presenting data in a table, there are some fields which have links by way of an InkWell. These InkWells seem to cause the reader to read out of order, being read last.

After much consultation of documentation, I am also unaware of how to change this ordering properly or group a row of data together. Any advice is greatly appreciated.

**Comment by chrisnorman7** [comment]
@UNIcodeX,
I've not used the InkWell class myself, but I am blind, and suspect I have done more than my fair share of getting the Semantics widget to play nice with screen readers. If I can be of help, please feel free to get in touch with me privately, and I'll see what I can do.

To answer your question, I do not know of a way to change the tab order, which is what it sounds like you want to do. I don't really want to derail this thread, but maybe have a think if InkWell is really what you want to use, or whether you could get away with a Column widget, or just make more rows in your table.

HTH.

**Comment by UNIcodeX** [comment]
@chrisnorman7 ,

Thank you for this offer. I'll email you a link to a gist I've prepared at the address listed on your github profile page. 

The InkWell widgets are being used for a clickable way for users to pass phone numbers and email addresses off to a URI handler, which makes use of the url_launcher pub.dev package. If there is another way to handle this which works to preserve proper tab ordering, as well as the ability to click the numbers and email addresses, then I'm all for it.
